scripts/globals/aman_trove.lua

```diff
@@ -0,0 +1,210 @@
+end
+
+local getLoot = function(battlefield)
+    return {
+        smallLoot  = battlefield:getLocalVar('SmallLoot'),
+        mediumLoot = battlefield:getLocalVar('MediumLoot'),
+        largeLoot  = battlefield:getLocalVar('LargeLoot'),
+    }
+end
+
+-----------------------------------
+-- Global Functions
+-----------------------------------
+xi.amantrove.setupBattlefield = function(battlefield)
+    local zone = battlefield:getZone()
+    local ID   = zones[zone:getID()]
+
+    -- Setup Terminal Coffer
+    local battleArea     = battlefield:getArea()
+    local terminalCoffer = GetNPCByID(ID.npc.TERMINAL_COFFER + (battleArea - 1) * 11)
+
+    -- NPC
+    terminalCoffer:setStatus(xi.status.NORMAL)
+    terminalCoffer:setUntargetable(false)
+
+    -- Setup Chests (mobs)
+    -- We pre-allocate everything rather than rolling on things at trigger-time.
+    local mimicChest = math.random(0, 9)
+
+    for i = 0, 9 do
+        local chest = GetMobByID(ID.mob.CHEST_O_PLENTY + i + (battleArea - 1) * 11)
+
+        chest:setStatus(xi.status.NORMAL) -- Make mob triggerable
+        chest:setUntargetable(false)
+
+        if i == mimicChest then
+            chest:setLocalVar('Mimic', 1)
+        end
+    end
+end
+
+xi.amanTrove.chestOPlentyOnTrigger = function(player, npc)
+    -- Debouncing (just in case)
+    if npc:getLocalVar('Triggered') == 1 then
+        return
+    end
+    npc:setLocalVar('Triggered', 1)
+
+    if npc:getLocalVar('Mimic') == 1 then
+        xi.amanTrove.prepareMimic(player, npc)
+        return
+    end
+
+    local zone = player:getZone()
+    local ID   = zones[zone:getID()]
+
+    print("You have triggered the Chest O'Plenty.")
+
+    player:showText(npc, ID.text.LOUD_THUD)
+    addSmallLoot(player:getBattlefield())
+
+    npc:setAnimationSub(1)
+    npc:setUntargetable(true)
+
+    -- TODO: Thuds etc.
+end
+
+xi.amanTrove.terminalCofferOnTrigger = function(player, npc)
+    -- Debouncing (just in case)
+    if npc:getLocalVar('Triggered') == 1 then
+        return
+    end
+    npc:setLocalVar('Triggered', 1)
+
+    npc:setAnimationSub(1)
+    npc:setUntargetable(true)
+
+    print("You have triggered the Terminal Coffer.")
+
+    -- TODO: Count up thuds, etc. generate loot
+    local loot = getLoot(player:getBattlefield())
+
+    -- TODO: We're going to predetermine how many thuds etc. there are
+    --     : available at the start of the BC. If somehow we get to this point
+    --     : and there are more than we allocated - then the player has managed
+    --     : to cheat and we should report it and make sure they don't get any
+    --     : loot!
+end
+
+xi.amanTrove.prepareMimic = function(player, npc)
+    print("The chest is a mimic!")
+
+    npc:setStatus(xi.status.UPDATE)
+    npc:setModelId(258)
+    npc:setAnimation(1) -- Combat
+    npc:setAnimationSub(1) -- Mouth open
+
+    -- Make things disappear:
+    -- npc:entityAnimationPacket('kesu', npc)
```

Comment:
We have this packet names enum'd.
Instead of `'kesu'` use `xi.animationString.STATUS_DISAPPEAR`